### PR TITLE
Gutenberg for Events Post Type

### PR DESCRIPTION
**Pull request by bordoni**
As previously discussed internally by our team we are going to start implementing Gutenberg into The Events Calendar. 

## Update 2018-01-16
This pull request is a first take of code to accomplish that, let's see how it evolves this week.

Doing [Babel compile](https://babeljs.io/docs/usage/cli/) for the only block we have manually with the following command:
```
$ npx babel src/resources/js/block/event\-details.raw.js --out-file src/resources/js/block/event\-details.js --presets=react
```
_We are using the [React preset](https://babeljs.io/docs/plugins/preset-react/)._

## Update 2018-01-17
Using a third-party tool to convert HTML into React Components after fetching it via the AJAX query. Using the same rendering method for both sides of the Blocks, will change later on.

Changed the folder in which I am using Babel CLI on:
```
$ npx babel src/resources/js/editor/blocks/event\-details.raw.js --out-file src/resources/js/editor/blocks/event\-details.js --presets=react
```

### CODE REVIEW

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:warning: | Comment refers to a TODO task | [Editor.php:13](https://github.com/moderntribe/the-events-calendar/blob/a292b8da1da23ead8b432fa941c9cc61e9d3441c/src/Tribe/Editor.php#L13)
:warning: | Comment refers to a TODO task | [Editor.php:62](https://github.com/moderntribe/the-events-calendar/blob/a292b8da1da23ead8b432fa941c9cc61e9d3441c/src/Tribe/Editor.php#L62)
:warning: | Comment refers to a TODO task | [Provider.php:23](https://github.com/moderntribe/the-events-calendar/blob/a292b8da1da23ead8b432fa941c9cc61e9d3441c/src/Tribe/Editor/Provider.php#L23)

> via jenkins/codesniffer-phpcs

File: src/Tribe/Editor/Provider.php [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+	 * Binds and sets up implementations.
+	 *
+	 * @since  TBD
+	 *
+	 */
+	public function register() {
+		$this->container->singleton( 'tec.editor', 'Tribe__Events__Editor' );
+
+		// Should we continue loading?
+		if ( ! tribe( 'tec.editor' )->is_gutenberg_active() ) {
```
Comment (zbtirrell) [inline src/Tribe/Editor/Provider.php]: should we also limit our scope to being on a TEC editor screen?

Comment (bordoni) [reply src/Tribe/Editor/Provider.php]: Yes and No. Parts of it need to load on the front-end for rendering blocks.

Comment (faction23) [reply src/Tribe/Editor/Provider.php]: Ya this is the interesting part of this game, especially as we get into complex blocks. Organizing code, especially the pcss/js that drives a block on fe and in edit mode in an efficient way is not something i have fully sorted yet on the agency side. working that through this week.

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:warning: | Comment refers to a TODO task | [Editor.php:13](https://github.com/moderntribe/the-events-calendar/blob/102e9fbc6c8fb58ed054c3b078bf3a10597b3b6d/src/Tribe/Editor.php#L13)
:warning: | Comment refers to a TODO task | [Editor.php:62](https://github.com/moderntribe/the-events-calendar/blob/102e9fbc6c8fb58ed054c3b078bf3a10597b3b6d/src/Tribe/Editor.php#L62)
:warning: | Comment refers to a TODO task | [Provider.php:23](https://github.com/moderntribe/the-events-calendar/blob/102e9fbc6c8fb58ed054c3b078bf3a10597b3b6d/src/Tribe/Editor/Provider.php#L23)

> via jenkins/codesniffer-phpcs

**Comment by bordoni** [comment]
We are closing this Pull Request in favor of moving Gutenberg code into an Extension repository.
https://github.com/moderntribe/events-gutenberg

**Pull request closed without merging**